pageserver/src/tenant/timeline.rs

```diff
@@ -365,7 +356,8 @@ pub struct Timeline {
     pub walreceiver: Mutex<Option<WalReceiver>>,
 
     /// Relation size cache
-    pub(crate) rel_size_cache: RwLock<RelSizeCache>,
+    pub(crate) rel_size_latest_cache: RwLock<HashMap<RelTag, (Lsn, BlockNumber)>>,
+    pub(crate) rel_size_snapshot_cache: Mutex<LruCache<(Lsn, RelTag), BlockNumber>>,
```

Comment:
Let's make this a read write lock too: `RwLock`.

Reply:
No sense: it is LRU cache, it means that it is updated on each access. So even get method is mutable.

Reply:
Yet another reason to not touch it from the primary :)